### PR TITLE
Dismiss notifications properly

### PR DESCRIPTION
**Pull request by rullzer**
Requires: https://github.com/nextcloud/server/pull/18297

### CODE REVIEW

**Comment by nickvergessen** [comment]
But hooking in `delete(` could run in circles... I will have a look later tonight

**Comment by rullzer** [comment]
> But hooking in `delete(` could run in circles... I will have a look later tonight

Delete calls deleteById so all roads pipe trough there if I'm not mistaken.

**Comment by nickvergessen** [comment]
Which in return means we are grabbing the data too many times, but yeah you can continue like this for now. I will improve the duplicated queries later.

**Comment by nickvergessen** [comment]
Pushed a fix

**Comment by nickvergessen** [comment]
I guess unit tests need adjusting

File: lib/Handler.php [hunk 1]
```diff
@@ -132,11 +134,20 @@ public function deleteByUser(string $user): bool {
 	/**
 	 * Delete the notification matching the given id
 	 *
+	/**
```
Comment (nickvergessen) [inline lib/Handler.php]: :man_facepalming:

**Pull request merged**